.github/workflows/ci.yaml

```diff
@@ -18,6 +18,21 @@ env:
   RUSTUP_MAX_RETRIES: 10
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      typescript: ${{ steps.filter.outputs.typescript }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
```

Comment:
Better to add commit hash for third party actions, otherwise this is too vulnerable to supply chain attack (someone might hack dorny and release bad `paths-filter@v2.1`, which would be propagated to us automatically. We don't do this for `actions/` actions cause we trust GitHub anyway)

Reply:
Good point